src/modules/cubedos-time_server-messages.adb

```diff
@@ -21,13 +21,21 @@ package body CubedOS.Time_Server.Messages
    -- messages to the tick generator and the task that sends tick messages to other modules.
    -- The tick generator will not need to send ticks to itself, so start at 2.
 
-   subtype Restricted_Module_ID is Positive range 2 .. Module_ID_Type'Last;
+   subtype Restricted_Module_ID is Module_ID_Type
+     with Dynamic_Predicate => Restricted_Module_ID /= Name_Resolver.Time_Server.Module_ID;
 
-   -- Should Series_Record hold the Domain_ID?
+   subtype Restricted_External_Address is Message_Address
+     with Dynamic_Predicate => not (
+                                 Restricted_External_Address.Module_ID = Name_Resolver.Time_Server.Module_ID
+                               and Restricted_External_Address.Domain_ID = Name_Resolver.Time_Server.Domain_ID);
+
+   -- Stores all persistent info about a series.
    type Series_Record is
       record
-         Module_ID : Restricted_Module_ID := Restricted_Module_ID'First;
-         ID        : Series_ID_Type := Series_ID_Type'First;
+         -- Doesn't matter what default values are, they should all be overriden.
+         -- We only have them because SPARK can't cope with uninitialized things.
+         Address : Restricted_External_Address := (0, 1);
```

Comment:
I think this will cause the dynamic predicate to fail should the time server happen to have the address (0, 1). As an aside, the domain IDs in the check program's name resolver should probably be set to 1 (i.e., a real domain number) rather than the placeholder value of 0 as they are now. I'm not sure of the special domain ID value of 0 is really very useful. Perhaps it shouldn't be allowed.

A bigger question is if the notion of restricted addresses (formerly restricted module IDs) are really worth the trouble? It may make sense to dynamically check that a request is not coming from the time server itself, but trying to do that with types might be the wrong approach.